### PR TITLE
Edit existing availability

### PR DESCRIPTION
**Pull request by ZhenyuYu1**
New Features:
1. When a user is editing their availability, the grid will be populated with their availability so they can see their own availability while editing.

Before:
Screenshot 2024-09-12 at 6 43 34 PM

Right after pressing "Edit Availability":
Screenshot 2024-09-12 at 6 44 15 PM

After changing availabilities, but still in edit mode:
Screenshot 2024-09-12 at 6 44 55 PM

After pressing save:
Screenshot 2024-09-12 at 6 45 20 PM

Another user's view of editing only their own availability:
Screenshot 2024-09-12 at 6 46 27 PM
Screenshot 2024-09-12 at 6 46 54 PM

### CODE REVIEW

File: src/components/AvailabilityGrid.tsx [hunk 1]
```diff
@@ -115,25 +115,64 @@ const Grid = ({
             const startIndex = timeArray.indexOf(timeSlot.beginning)
             let endIndex = timeArray.indexOf(timeSlot.end)
 
-            // Handle case where end time is not found in the timeArray (endIndex being -1)
-            if (endIndex === -1) {
-              endIndex = timeArray.length // Set to the end of the timeArray
+            // Instead of just setting to true, increment a counter to keep track of how many people are available at that time
```
Comment (pselvana) [inline src/components/AvailabilityGrid.tsx:118]: can you confirm this code is what you added and not a revert or a re-add of previous code please? I feel like this function keeps going back and forth from previous changes.

File: src/components/AvailabilityGrid.tsx [hunk 2]
```diff
@@ -335,9 +375,7 @@ const Grid = ({
 
           {/* Main Grid body cells for selecting and deselecting */}
           <div
-            className={`grid h-full border border-gray-300 ${
-              isAvailable ? 'bg-red-50' : ''
-            }`}
+            className={`grid h-full border border-gray-300`}
```
Comment (pselvana) [inline src/components/AvailabilityGrid.tsx:378]: is this an expected styling difference?

**Review by pselvana — APPROVED** [review]
LGTM!

File: src/components/AvailabilityGrid.tsx (hunk 1 above)
Comment (ZhenyuYu1) [reply src/components/AvailabilityGrid.tsx:118]: The current version should be correct. I think it got reverted in a previous merge, but now it should be correct.

**Pull request merged**